controllers/vspherevm_ipaddress_reconciler.go

```diff
@@ -112,21 +121,51 @@ func (r vmReconciler) reconcileIPAddressClaims(ctx context.Context, vmCtx *capvc
 			claims,
 			conditions.AddSourceRef(),
 			conditions.WithStepCounter())
+
+		if len(v1beta2Claims) > 0 {
+			if err := v1beta2conditions.SetAggregateCondition(v1beta2Claims, vmCtx.VSphereVM, clusterv1.ReadyV1Beta2Condition, v1beta2conditions.TargetConditionType(infrav1.VSphereVMIPAddressClaimsFulfilledV1Beta2Condition)); err != nil {
+				return errors.Wrap(err, "failed to aggregate Ready condition from IPAddress claims")
+			}
+		} else {
+			v1beta2conditions.Set(vmCtx.VSphereVM, metav1.Condition{
+				Type:    infrav1.VSphereVMIPAddressClaimsFulfilledV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
```

Comment:
Should this be true?

I think we only hit this case if a VSphereVM has no AddressesFromPools

I think in this case we don't want to set the VSphereVM ready condition to false, right?